src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory.vb

```diff
@@ -1765,6 +1765,32 @@ Namespace Microsoft.CodeAnalysis.Operations
                 Return New LazyInvalidOperation(children, _semanticModel, syntax, type, constantValue, isImplicit)
             End If
         End Function
+
+        Private Function CreateBoundReDimOperation(boundRedimStatement As BoundRedimStatement) As IReDimOperation
+            Dim preserve As Boolean = boundRedimStatement.Syntax.Kind = SyntaxKind.ReDimPreserveStatement
```

Comment:
Clauses can be empty for error cases. Do we fallback to this implementation for empty clauses? I thought this implementation is fine given the assert below that confirms the value matches the preserve value stored in each clause bound node.

---
In reply to: [221787027](https://github.com/dotnet/roslyn/pull/30217#discussion_r221787027) [](ancestors = 221787027)

Reply:
> Clauses can be empty for error cases. Do we fallback to this implementation for empty clauses? I thought this implementation is fine given the assert below that confirms the value matches the preserve value stored in each clause bound node.

In general we prefer getting information from bound nodes rather than inferring it from syntax.

---
In reply to: [223452132](https://github.com/dotnet/roslyn/pull/30217#discussion_r223452132) [](ancestors = 223452132,221787027)